src/ui/public/vis/request_handlers/courier.js

```diff
@@ -133,7 +134,8 @@ const CourierRequestHandlerProvider = function () {
 
       const parsedTimeRange = timeRange ? getTime(aggs.indexPattern, timeRange) : null;
       const tabifyParams = {
-        metricsAtAllLevels: isHierarchical,
+        isHierarchical,
+        metricsAtAllLevels,
```

Comment:
These are split into separate params, because within table vis there is a distinction between the two. For example, the following is possible:
```javascript
vis.params.showMetricsAtAllLevels === false;
vis.params.showPartialRows === true;
vis.isHierarchical(); // still returns true
```

This is due to how we configure the table vis:
 https://github.com/elastic/kibana/blob/9b08fc719a9fb4c0c1dce55be73d395e7cd6b54a/src/core_plugins/table_vis/public/table_vis.js#L114-L116

Reply:
this is actually wrong i think .... 
`hierarchical === metricsAtAllLevels` (unless someone knows of any reasons it should be ? )

in courier request handler vis.isHierarchical() is used to figure out if we need to build a request with metrics on every level. (or it was, and its now refactored to look at metricsAtAllLevels parameter). we should use this same parameter in tabify (to figure out the same thing) ...
and partialRows should be completely independent ... as its just removing rows where we don't have value for every column.